File: src/main/java/j/chain_of_responsibility/like/functions/MessagePrinter.java
package j.chain_of_responsibility.like.functions;

import j.chain_of_responsibility.Trouble;

public final class MessagePrinter {
  public static void done(String name, Trouble trouble) {
    System.out.println(trouble + " is resolved by [" + name + "].");
  }

  public static void fail(Trouble trouble) {
    System.out.println(trouble + " cannot be resolved.");
  }
}

File: src/main/scala/s/chain_of_responsibility/like/functional/MessagePrinter.scala
package s.chain_of_responsibility.like.functional

import s.chain_of_responsibility.Trouble

object MessagePrinter {
  def done(name: String, trouble: Trouble): Unit = {
    System.out.println(trouble.toString + " is resolved by [" + name + "].")
  }

  def fail(trouble: Trouble): Unit = {
    System.out.println(trouble.toString + " cannot be resolved.")
  }
}
